pkg/controller/service/service_controller.go

```diff
@@ -568,8 +568,9 @@ func portEqualForLB(x, y *v1.ServicePort) bool {
 		return false
 	}
 
-	// We don't check TargetPort; that is not relevant for load balancing
-	// TODO: Should we blank it out?  Or just check it anyway?
+	if !reflect.DeepEqual(x.TargetPort, y.TargetPort) {
```

Comment:
I think you should be able to do:

`if x.TargetPort != y.TargetPort`

Reply:
the struct of targetPort is `TargetPort intstr.IntOrString`. and I saw that the author to check status change or not by `reflect.DeepEqual` in the other place of code.

Reply:
It seems to be a struct of all non pointer types, so it should be straightforwardly comparable? Use of reflect is usually pretty heavyweight...

Reply:
reasonable, I will fix it.